libs/utils/decorate.js

```diff
@@ -19,13 +19,28 @@ let videoLabels = {
 };
 let videoCounter = 0;
 
+const shouldBlockFreeTrialLinks = (a, prefix, parent) => {
+  if (prefix !== '/kr' || (!a.dataset?.modalPath?.includes('/kr/cc-shared/fragments/trial-modals')
+    && !['free-trial', 'free trial', '무료 체험판', '무료 체험하기',
+      '{{free-trial}}', '{{start-free-trial}}', '{{try-for-free}}']
+      .some((pattern) => a.textContent?.toLowerCase()?.includes(pattern.toLowerCase())))) {
+    return false;
+  }
+
+  const elementToRemove = (parent.tagName === 'STRONG' || parent.tagName === 'EM') && parent.children.length === 1 ? parent : a;
+  elementToRemove.remove();
+  return true;
+};
+
 export function decorateButtons(el, size) {
   const buttons = el.querySelectorAll('em a, strong a, p > a strong');
   if (buttons.length === 0) return;
   const buttonTypeMap = { STRONG: 'blue', EM: 'outline', A: 'blue' };
+
   buttons.forEach((button) => {
-    let target = button;
     const parent = button.parentElement;
+    if (shouldBlockFreeTrialLinks(button, getConfig()?.locale?.prefix, parent)) return;
```

Comment:
My rule of thumb is: 2+ parms -> use object, it's usually easier to expand. However, I'd like to challenge having `getConfig()?.locale?.prefix` here and not in the method itself. Is there an advantage to this? If we were to cache it outside of the method call, then we'd at least avoid calling the same method for each button; but currently that's not the case. So let's either move the prefix definition to the method itself or declare it outside of the `buttons` loop and use an object as a param.